src/redactableProcessor.js

```diff
@@ -70,13 +108,23 @@ module.exports = class RedactableProcessor {
     return mergedTree;
   }
 
-  sourceAndRedactedToRestored(source, redacted) {
+  sourceAndRedactedToRestored(source, redacted, strict) {
     const sourceTree = this.sourceToRedactedSyntaxTree(source);
-    const restorationTree = this.redactedToSyntaxTree(redacted);
+    const restorationTree = this.redactedToSyntaxTree(redacted, strict);
     const mergedSyntaxTree = this.sourceAndRedactedToMergedSyntaxTree(
       sourceTree,
-      restorationTree
+      JSON.parse(JSON.stringify(restorationTree))
```

Comment:
what's this for?

Reply:
The transformer modifies the tree, which means that the verification logic can't rely on information from the `restorationTree`.

I found this method of deep copying in a couple of place ([example](https://www.codementor.io/ramnmiklus/deep-copying-an-object-in-javascript-mdlj2c318#a-solution)) but definitely open to suggestions of a better way. I'll add a comment here at the very least.

Reply:
Oh, true. I don't love that that's a side effect. I'd vote we rewrite the transformer to use something like https://github.com/syntax-tree/unist-util-map to create a new tree. Alternatively, the "deep copy" logic should probably happen in the `sourceAndRedactedToMergedSyntaxTree` method itself

Reply:
Yeah I don't love the side effect either. But, I remember trying to use `map` instead of `visit` and it caused all sorts of problems with block restorations because we might change multiple blocks at once.